src/ocl/convolutionocl.cpp

```diff
@@ -655,49 +674,66 @@ static void DirConvFindCore(Handle& handle,
+    // Find all sollutions before parallel compiling these solutions.
+    const auto winograd_solutions =
+        !use_winograd_only ? conv.FindWinogradSolutions(ctx, invoke_ctx) : [&]() {
             AutoUseFastDynamicSolutions tmp{ctx};
             return conv.FindWinogradSolutions(ctx, invoke_ctx);
         }();
-        PrecompileSolutions(handle, all);
-        const auto algorithm_name = AlgorithmName{"miopenConvolutionFwdAlgoWinograd"};
-        EvaluateInvokers(handle, all, algorithm_name, network_config, invoke_ctx, record);
-    }
-
-    // Direct algo
-    if(!use_winograd_only)
-    {
-        ConvolutionUserBuffers bufs(workSpace, workSpaceSize);
-        bufs.SetFwd(x, w, y);
-        const auto all = conv.FindDataDirectSolutions(
-            handle, xDesc, wDesc, yDesc, exhaustiveSearch, true, bufs, invoke_ctx);
-        PrecompileSolutions(handle, all);
-        const auto algorithm_name = AlgorithmName{"miopenConvolutionFwdAlgoDirect"};
-        EvaluateInvokers(handle, all, algorithm_name, network_config, invoke_ctx, record);
-    }
-
-    // Implicit GEMM algo
-    if(!use_winograd_only)
-    {
-        ConvolutionUserBuffers bufs(workSpace, workSpaceSize);
-        bufs.SetFwd(x, w, y);
-        const auto all = conv.FindDataImplicitGemmSolutions(
-            handle, xDesc, wDesc, yDesc, exhaustiveSearch, true, bufs, invoke_ctx);
-        PrecompileSolutions(handle, all);
-        const auto algorithm_name = AlgorithmName{"miopenConvolutionFwdAlgoImplicitGEMM"};
-        EvaluateInvokers(handle, all, algorithm_name, network_config, invoke_ctx, record);
-    }
-
-    // FFT algo
-    if(!use_winograd_only && !miopen::IsDisabled(MIOPEN_DEBUG_CONV_FFT{}))
-    {
-        const auto all            = FindAllFFTSolutions(ctx, invoke_ctx);
-        const auto algorithm_name = AlgorithmName{"miopenConvolutionFwdAlgoFFT"};
-        PrecompileSolutions(handle, all);
-        EvaluateInvokers(handle, all, algorithm_name, network_config, invoke_ctx, record);
-    }
+    ConvolutionUserBuffers bufs(workSpace, workSpaceSize);
+    bufs.SetFwd(x, w, y);
+    const auto direct_solutions =
+        !use_winograd_only
+            ? conv.FindDataDirectSolutions(
+                  handle, xDesc, wDesc, yDesc, exhaustiveSearch, true, bufs, invoke_ctx)
+            : std::vector<miopen::solver::ConvSolution>{};
+    const auto implictgemm_solutions =
+        !use_winograd_only
+            ? conv.FindDataImplicitGemmSolutions(
+                  handle, xDesc, wDesc, yDesc, exhaustiveSearch, true, bufs, invoke_ctx)
+            : std::vector<miopen::solver::ConvSolution>{};
+    const auto fft_solutions = !use_winograd_only ? conv.FindFftSolutions(ctx, invoke_ctx)
+                                                  : std::vector<miopen::solver::ConvSolution>{};
+
+    // Precompile Solutions
+    {
+        std::vector<const miopen::solver::ConvSolution*> all_solutions;
```

Comment:
[Recommendation] [Style] Remove `_solutions` suffix everywhere. It seems obvious (to me) that `direct_solutions`, `all_solutions` etc are some containers of solutions.